diff -aur openvdb-7.1.0/cmake/OpenVDBGLFW3Setup.cmake openvdb-7.1.0.new/cmake/OpenVDBGLFW3Setup.cmake
--- openvdb-7.1.0/cmake/OpenVDBGLFW3Setup.cmake	2020-08-13 21:15:58.000000000 +0200
+++ openvdb-7.1.0.new/cmake/OpenVDBGLFW3Setup.cmake	2020-12-12 16:03:19.304427600 +0100
@@ -109,11 +109,17 @@
 set(glfw3_FIND_VERSION ${MINIMUM_GLFW_VERSION})
 find_package(glfw3 ${MINIMUM_GLFW_VERSION} REQUIRED)
 
-find_package(PackageHandleStandardArgs)
-find_package_handle_standard_args(glfw3
-  REQUIRED_VARS glfw3_DIR glfw3_FOUND
-  VERSION_VAR glfw3_VERSION
-)
+# We only use find_package_handle_standard_args to verify and print
+# appropriate messages. This now explicitly errors in 3.19...
+# @todo Improve this entire GLFW3 search
+#  https://gitlab.kitware.com/cmake/cmake/-/issues/21505
+if(${CMAKE_VERSION} VERSION_LESS 3.19)
+  find_package(PackageHandleStandardArgs)
+  find_package_handle_standard_args(glfw3
+    REQUIRED_VARS glfw3_DIR glfw3_FOUND
+    VERSION_VAR glfw3_VERSION
+  )
+endif()
 
 unset(glfw3_FIND_VERSION)
 
