diff -aur OpenUSD-24.08/pxr/base/work/dispatcher.cpp OpenUSD-24.08.new/pxr/base/work/dispatcher.cpp
--- OpenUSD-24.08/pxr/base/work/dispatcher.cpp	2024-07-25 00:34:34.000000000 +0200
+++ OpenUSD-24.08.new/pxr/base/work/dispatcher.cpp	2024-12-03 14:39:40.850677580 +0100
@@ -33,7 +33,11 @@
 #if TBB_INTERFACE_VERSION_MAJOR >= 12
 inline tbb::detail::d1::wait_context& 
 WorkDispatcher::_TaskGroup::_GetInternalWaitContext() {
+#if TBB_INTERFACE_VERSION_MINOR >= 14
+    return m_wait_vertex.get_context();
+#else
     return m_wait_ctx;
+#endif
 }
 #endif
 
