diff -aur OpenUSD-24.08/pxr/base/work/dispatcher.h OpenUSD-24.08.new/pxr/base/work/dispatcher.h
--- OpenUSD-24.08/pxr/base/work/dispatcher.h	2024-07-25 00:34:34.000000000 +0200
+++ OpenUSD-24.08.new/pxr/base/work/dispatcher.h	2024-11-24 14:45:10.426650090 +0100
@@ -21,6 +21,7 @@
 #include <tbb/blocked_range.h>
 #include <tbb/concurrent_vector.h>
 #if TBB_INTERFACE_VERSION_MAJOR >= 12
+#include <tbb/collaborative_call_once.h>
 #include <tbb/task_group.h>
 #else
 #include <tbb/task.h>
@@ -255,4 +256,4 @@
 
 PXR_NAMESPACE_CLOSE_SCOPE
 
-#endif // PXR_BASE_WORK_DISPATCHER_H
\ Pas de fin de ligne à la fin du fichier
+#endif // PXR_BASE_WORK_DISPATCHER_H
