diff -aur blender/source/blender/python/generic/python_compat.cc blender.new/source/blender/python/generic/python_compat.cc
--- blender/source/blender/python/generic/python_compat.cc	2025-10-12 11:48:44.000000000 +0200
+++ blender.new/source/blender/python/generic/python_compat.cc	2025-10-13 14:07:23.438709169 +0200
@@ -14,9 +14,9 @@
 #include "BLI_utildefines.h" /* IWYU pragma: keep. */
 #include "python_compat.hh"  /* IWYU pragma: keep. */
 
-#if PY_VERSION_HEX >= 0x030e0000 /* >=3.14 */
+#if PY_VERSION_HEX >= 0x030d0000 /* >=3.13 */
 
-/* Removed in Python 3.14. */
+/* Not declared in this scope in Python 3.13. */
 int _PyArg_CheckPositional(const char *name, Py_ssize_t nargs, Py_ssize_t min, Py_ssize_t max)
 {
   BLI_assert(min >= 0);
Seulement dans blender.new/source/blender/python/generic: python_compat.cc~
diff -aur blender/source/blender/python/generic/python_compat.hh blender.new/source/blender/python/generic/python_compat.hh
--- blender/source/blender/python/generic/python_compat.hh	2025-10-12 11:48:44.000000000 +0200
+++ blender.new/source/blender/python/generic/python_compat.hh	2025-10-13 14:09:25.835991210 +0200
@@ -49,6 +49,5 @@
 #  define Py_HashPointer _Py_HashPointer
 #  define PyThreadState_GetUnchecked _PyThreadState_UncheckedGet
 /* TODO: Support: `PyDict_Pop`, it has different arguments. */
-#else /* >= 3.14 */
 int _PyArg_CheckPositional(const char *name, Py_ssize_t nargs, Py_ssize_t min, Py_ssize_t max);
 #endif
