diff -aur blender/source/blender/blenkernel/intern/customdata.cc blender.new/source/blender/blenkernel/intern/customdata.cc
--- blender/source/blender/blenkernel/intern/customdata.cc	2026-03-01 11:09:43.000000000 +0100
+++ blender.new/source/blender/blenkernel/intern/customdata.cc	2026-03-01 11:24:51.326031759 +0100
@@ -209,7 +209,7 @@
   for (i = 0; i < count; i++) {
     MDeformVert *dvert = static_cast<MDeformVert *>(POINTER_OFFSET(dest, i * size));
 
-    if (dvert->totweight) {
+    if (dvert->totweight > 0) {
       MDeformWeight *dw = MEM_new_array_uninitialized<MDeformWeight>(size_t(dvert->totweight),
                                                                      __func__);
 
@@ -218,6 +218,7 @@
     }
     else {
       dvert->dw = nullptr;
+      dvert->totweight = 0;
     }
   }
 }
