diff -aur oiio/src/include/OpenImageIO/simd.h oiio.new/src/include/OpenImageIO/simd.h
--- oiio/src/include/OpenImageIO/simd.h	2016-10-01 16:09:45.000000000 +0200
+++ oiio.new/src/include/OpenImageIO/simd.h	2016-10-02 10:59:35.362404176 +0200
@@ -2325,7 +2325,7 @@
 # if OIIO_SIMD_AVX >= 2 && (OIIO_GNUC_VERSION > 50000)
     // Fastest way to fill with all 1 bits is to cmp any value to itself.
     __m256i anyval = _mm256_undefined_si256();
-    return _mm256_cmpeq_epi8 (anyval, anyval);
+    return _mm256_castsi256_ps (_mm256_cmpeq_epi8 (anyval, anyval));
 # else
     return _mm256_castsi256_ps (_mm256_set1_epi32 (-1));
 # endif
