Index: konversation/trunk/PKGBUILD
===================================================================
--- konversation/trunk/PKGBUILD	(revision 40236)
+++ konversation/trunk/PKGBUILD	(working copy)
@@ -4,19 +4,26 @@
 
 pkgname=konversation
 pkgver=1.2.alpha2
+_pkgver=1.2-alpha2
 pkgrel=1
 pkgdesc="A user friendly IRC client for KDE"
 arch=(i686 x86_64)
 url="http://konversation.kde.org"
-license="GPL"
-depends=('kdelibs' 'libxss' 'kdepimlibs' 'qt')
-makedepends=('automoc4' 'cmake' 'pkgconfig')
-source=(http://download.berlios.de/$pkgname/$pkgname-1.2-alpha2.tar.bz2)
+license=('GPL2' 'FDL')
+depends=('kdepimlibs' 'libxss')
+makedepends=('automoc4' 'cmake')
+source=(http://download.berlios.de/${pkgname}/${pkgname}-${_pkgver}.tar.bz2)
 
 build() {
-  cd $srcdir/$pkgname-1.2-alpha2
-  cmake -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_BUILD_TYPE=Release
-  make || return 1
-  make DESTDIR=$startdir/pkg install
+	cd ${srcdir}
+	mkdir build
+	cd build
+	cmake ../${pkgname}-${_pkgver} \
+		-DCMAKE_BUILD_TYPE=Release \
+		-DCMAKE_SKIP_RPATH=ON \
+		-DCMAKE_{SHARED,MODULE,EXE}_LINKER_FLAGS='-Wl,--no-undefined -Wl,--as-needed' \
+		-DCMAKE_INSTALL_PREFIX=/usr
+	make || return 1
+	make DESTDIR=${pkgdir} install || return 1
 }
 md5sums=('213a96650b9493fe650e44c4bddc7b27')
