diff -aur conky/src/luamm.cc conky.new/src/luamm.cc
--- conky/src/luamm.cc	2015-12-06 15:40:27.000000000 +0100
+++ conky.new/src/luamm.cc	2017-03-06 15:29:29.462699931 +0100
@@ -165,7 +165,14 @@
 		}
 
 		template<void (*misc)(lua_State *, int), int nresults>
-		int safe_misc_trampoline(lua_State *l)
+		int safe_misc_trampoline_set(lua_State *l)
+		{
+			misc(l, 1);
+			return nresults;
+		}
+
+		template<int (*misc)(lua_State *, int), int nresults>
+		int safe_misc_trampoline_get(lua_State *l)
 		{
 			misc(l, 1);
 			return nresults;
@@ -334,7 +341,7 @@
 		if( rawequal(index1, index2) )
 			return true;
 
-		return safe_compare(&safe_compare_trampoline<lua_equal>, index1, index2);
+		return safe_compare(&safe_compare_trampoline<&lua_equal>, index1, index2);
 	}
 
 	int state::gc(int what, int data)
@@ -376,7 +383,7 @@
 		checkstack(2);
 		pushvalue(index);
 		insert(-2);
-		lua_pushcfunction(cobj.get(), (&safe_misc_trampoline<&lua_gettable, 1>));
+		lua_pushcfunction(cobj.get(), (&safe_misc_trampoline_get<&lua_gettable, 1>));
 		insert(-3);
 		call(2, 1, 0);
 	}
@@ -501,7 +508,7 @@
 		checkstack(2);
 		pushvalue(index);
 		insert(-3);
-		lua_pushcfunction(cobj.get(), (&safe_misc_trampoline<&lua_settable, 0>));
+		lua_pushcfunction(cobj.get(), (&safe_misc_trampoline_set<&lua_settable, 0>));
 		insert(-4);
 		call(3, 0, 0);
 	}
