diff -aur embree-3.5.2/common/cmake/gnu.cmake embree-3.5.2.new/common/cmake/gnu.cmake
--- embree-3.5.2/common/cmake/gnu.cmake	2019-03-13 12:13:56.000000000 +0100
+++ embree-3.5.2.new/common/cmake/gnu.cmake	2019-08-21 11:45:27.577667816 +0200
@@ -62,7 +62,7 @@
 
 SET(CMAKE_CXX_FLAGS_RELEASE "")
 SET(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -DNDEBUG")     # disable assertions
-SET(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O3")          # enable full optimizations
+SET(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -march=native -Ofast -pipe -fdiagnostics-color=always -fno-plt -floop-interchange -floop-strip-mine -floop-block")          # enable full optimizations
 SET(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -D_FORTIFY_SOURCE=2") # perform extra security checks for some standard library calls
 
 SET(CMAKE_CXX_FLAGS_RELWITHDEBINFO "")
