Seulement dans blender.new: blender
diff -aur blender/build_files/cmake/Modules/FindOpenEXR.cmake blender.new/build_files/cmake/Modules/FindOpenEXR.cmake
--- blender/build_files/cmake/Modules/FindOpenEXR.cmake	2020-08-19 12:06:57.000000000 +0200
+++ blender.new/build_files/cmake/Modules/FindOpenEXR.cmake	2021-04-28 14:25:24.778667825 +0200
@@ -34,11 +34,10 @@
 SET(_openexr_libs_ver_init "2.0")
 
 SET(_openexr_FIND_COMPONENTS
-  Half
-  Iex
-  IlmImf
-  IlmThread
-  Imath
+   Iex
+   Imath
+   OpenEXR
+   IlmThread
 )
 
 SET(_openexr_SEARCH_DIRS
@@ -120,7 +119,7 @@
 IF(OPENEXR_FOUND)
   SET(OPENEXR_LIBRARIES ${_openexr_LIBRARIES})
   # Both include paths are needed because of dummy OSL headers mixing #include <OpenEXR/foo.h> and #include <foo.h> :(
-  SET(OPENEXR_INCLUDE_DIRS ${OPENEXR_INCLUDE_DIR} ${OPENEXR_INCLUDE_DIR}/OpenEXR)
+  SET(OPENEXR_INCLUDE_DIRS ${OPENEXR_INCLUDE_DIR} ${OPENEXR_INCLUDE_DIR}/OpenEXR ${OPENEXR_INCLUDE_DIR}/Imath)
 ENDIF()
 
 MARK_AS_ADVANCED(
diff -aur blender/source/blender/imbuf/intern/openexr/openexr_api.cpp blender.new/source/blender/imbuf/intern/openexr/openexr_api.cpp
--- blender/source/blender/imbuf/intern/openexr/openexr_api.cpp	2021-04-11 10:34:57.000000000 +0200
+++ blender.new/source/blender/imbuf/intern/openexr/openexr_api.cpp	2021-04-28 14:31:09.689307246 +0200
@@ -38,8 +38,10 @@
 #include <ImfChannelList.h>
 #include <ImfCompression.h>
 #include <ImfCompressionAttribute.h>
+#include <ImfFrameBuffer.h>
 #include <ImfIO.h>
 #include <ImfInputFile.h>
+#include <ImfInt64.h>
 #include <ImfOutputFile.h>
 #include <ImfPixelType.h>
 #include <ImfStandardAttributes.h>
