diff -aur blender/intern/ffmpeg/ffmpeg_compat.h blender.new/intern/ffmpeg/ffmpeg_compat.h
--- blender/intern/ffmpeg/ffmpeg_compat.h	2024-10-03 18:27:32.000000000 +0200
+++ blender.new/intern/ffmpeg/ffmpeg_compat.h	2024-11-06 11:33:09.283443702 +0100
@@ -161,6 +161,15 @@
   return align;
 }
 
+FFMPEG_INLINE
+bool av_get_cur_key_frame_pts(const AVFrame *picture) {
+  #if LIBAVUTIL_VERSION_INT >= AV_VERSION_INT(58, 29, 100)
+  return (picture->flags & AV_FRAME_FLAG_KEY);
+  #else
+  return (picture->key_frame);
+  #endif
+}
+
 /* -------------------------------------------------------------------- */
 /** \name Deinterlace code block
  *
diff -aur blender/source/blender/imbuf/intern/anim_movie.cc blender.new/source/blender/imbuf/intern/anim_movie.cc
--- blender/source/blender/imbuf/intern/anim_movie.cc	2024-11-06 11:27:28.000000000 +0100
+++ blender.new/source/blender/imbuf/intern/anim_movie.cc	2024-11-06 11:36:53.905716785 +0100
@@ -700,7 +700,7 @@
 {
   anim->cur_pts = av_get_pts_from_frame(anim->pFrame);
 
-  if (anim->pFrame->key_frame) {
+  if (av_get_cur_key_frame_pts(anim->pFrame)) {
     anim->cur_key_frame_pts = anim->cur_pts;
   }
 
@@ -1079,11 +1079,10 @@
 
     AVFormatContext *format_ctx = anim->pFormatCtx;
 
-    if (format_ctx->iformat->read_seek2 || format_ctx->iformat->read_seek) {
-      ret = av_seek_frame(anim->pFormatCtx, anim->videoStream, seek_pos, AVSEEK_FLAG_BACKWARD);
-    }
-    else {
-      ret = ffmpeg_generic_seek_workaround(anim, &seek_pos, pts_to_search);
+    int ret = av_seek_frame(anim->pFormatCtx, anim->videoStream, seek_pos, AVSEEK_FLAG_BACKWARD);
+
+    if (ret < 0) {
+      ret = ffmpeg_generic_seek_workaround(anim, &seek_pos, pts_to_search);\
       av_log(anim->pFormatCtx,
              AV_LOG_DEBUG,
              "Adjusted final seek seek_pos = %" PRId64 "\n",
