diff -aur blender/intern/ffmpeg/ffmpeg_compat.h blender.new/intern/ffmpeg/ffmpeg_compat.h
--- blender/intern/ffmpeg/ffmpeg_compat.h	2024-11-20 14:13:20.000000000 +0100
+++ blender.new/intern/ffmpeg/ffmpeg_compat.h	2024-11-21 09:21:58.710574759 +0100
@@ -168,6 +168,15 @@
   return align;
 }
 
+FFMPEG_INLINE
+bool av_get_cur_key_frame_pts(const AVFrame *picture) {
+  #if LIBAVUTIL_VERSION_INT >= AV_VERSION_INT(58, 29, 100)
+  return (picture->flags & AV_FRAME_FLAG_KEY);
+  #else
+  return (picture->key_frame);
+  #endif
+}
+
 /* -------------------------------------------------------------------- */
 /** \name Deinterlace code block
  *
diff -aur blender/source/blender/imbuf/intern/anim_movie.cc blender.new/source/blender/imbuf/intern/anim_movie.cc
--- blender/source/blender/imbuf/intern/anim_movie.cc	2024-11-20 14:13:20.000000000 +0100
+++ blender.new/source/blender/imbuf/intern/anim_movie.cc	2024-11-21 09:27:56.978595575 +0100
@@ -705,7 +705,7 @@
   anim->cur_pts = av_get_pts_from_frame(anim->pFrame);
 
 #  ifdef FFMPEG_OLD_KEY_FRAME_QUERY_METHOD
-  if (anim->pFrame->key_frame) {
+  if (av_get_cur_key_frame_pts(anim->pFrame)) {
 #  else
   if (anim->pFrame->flags & AV_FRAME_FLAG_KEY) {
 #  endif
@@ -1027,11 +1027,10 @@
 
     AVFormatContext *format_ctx = anim->pFormatCtx;
 
-    if (format_ctx->iformat->read_seek2 || format_ctx->iformat->read_seek) {
-      ret = av_seek_frame(anim->pFormatCtx, anim->videoStream, seek_pos, AVSEEK_FLAG_BACKWARD);
-    }
-    else {
-      ret = ffmpeg_generic_seek_workaround(anim, &seek_pos, pts_to_search);
+    int ret = av_seek_frame(anim->pFormatCtx, anim->videoStream, seek_pos, AVSEEK_FLAG_BACKWARD);
+
+    if (ret < 0) {
+      ret = ffmpeg_generic_seek_workaround(anim, &seek_pos, pts_to_search);\
       av_log(anim->pFormatCtx,
              AV_LOG_DEBUG,
              "Adjusted final seek seek_pos = %" PRId64 "\n",
