diff -aur blender/release/scripts/addons/io_mesh_stl/blender_utils.py blender.new/release/scripts/addons/io_mesh_stl/blender_utils.py
--- blender/release/scripts/addons/io_mesh_stl/blender_utils.py	2017-11-12 16:26:23.000000000 +0100
+++ blender.new/release/scripts/addons/io_mesh_stl/blender_utils.py	2018-10-04 11:44:20.981305493 +0200
@@ -86,7 +86,7 @@
     try:
         mesh = ob.to_mesh(bpy.context.scene, use_mesh_modifiers, "PREVIEW")
     except RuntimeError:
-        raise StopIteration
+        return
 
     mat = global_matrix * ob.matrix_world
     mesh.transform(mat)
