diff -aur blender/source/blender/python/generic/py_capi_utils.hh blender.new/source/blender/python/generic/py_capi_utils.hh
--- blender/source/blender/python/generic/py_capi_utils.hh	2024-10-20 11:16:28.000000000 +0200
+++ blender.new/source/blender/python/generic/py_capi_utils.hh	2024-10-20 14:51:48.791295817 +0200
@@ -342,7 +342,7 @@
 #if PY_VERSION_HEX <= 0x030c0000 /* <=3.12 */
   return (int32_t)_PyLong_AsInt(value);
 #else
-  return (int32_t)PyLong_AsInt(value);
+  return (int32_t)_PyLong_AsInt(value);
 #endif
 }
 Py_LOCAL_INLINE(int64_t) PyC_Long_AsI64(PyObject *value)
