diff -aur blender/release/scripts/addons_contrib/curve_tools/CurveIntersections.py blender.new/release/scripts/addons_contrib/curve_tools/CurveIntersections.py
--- blender/release/scripts/addons_contrib/curve_tools/CurveIntersections.py	2016-04-03 11:00:13.000000000 +0200
+++ blender.new/release/scripts/addons_contrib/curve_tools/CurveIntersections.py	2017-01-09 17:36:26.258205426 +0100
@@ -470,6 +470,8 @@
     def InitGlobals():
         CurvesIntersector.ResetGlobals()
         
+        global algoDIR
+        global algoPOV
         
         algo = bpy.context.scene.curvetools.IntersectCurvesAlgorithm
         if algo == 'From View':
@@ -487,7 +489,7 @@
                 print("--", "viewMatrix:")
                 print(viewMatrix)
                 
-                global algoDIR
+                #global algoDIR
                 algoDIR = Vector((viewMatrix[2][0], viewMatrix[2][1], viewMatrix[2][2]))
                 print("--", "algoDIR:", algoDIR)
 
@@ -497,7 +499,7 @@
                 print("--", "viewMatrix:")
                 print(viewMatrix)
                 
-                global algoPOV
+                #global algoPOV
                 algoPOV = regionView3D.view_location.copy()
                 print("--", "algoPOV:", algoPOV)
                 
@@ -533,13 +535,13 @@
                 if cameraData.type == 'ORTHO':
                     cameraMatrix = camera.matrix_world
                 
-                    global algoDIR
+                    #global algoDIR
                     #algoDIR = Vector((cameraMatrix[2][0], cameraMatrix[2][1], cameraMatrix[2][2]))
                     algoDIR = Vector((- cameraMatrix[0][2], - cameraMatrix[1][2], - cameraMatrix[2][2]))
                     print("--", "algoDIR:", algoDIR)
 
                 if cameraData.type == 'PERSP':
-                    global algoPOV
+                    #global algoPOV
                     algoPOV = camera.location.copy()
                     print("--", "algoPOV:", algoPOV)
 
@@ -787,4 +789,4 @@
         
         
         
-        
\ Pas de fin de ligne à la fin du fichier
+        
