diff -aur blender/release/scripts/addons/node_wrangler.py blender.new/release/scripts/addons/node_wrangler.py
--- blender/release/scripts/addons/node_wrangler.py	2019-03-17 14:00:09.000000000 +0100
+++ blender.new/release/scripts/addons/node_wrangler.py	2019-03-17 14:07:40.193015546 +0100
@@ -4035,7 +4035,7 @@
             for vcol in vcols:
                 l.operator(NWAddAttrNode.bl_idname, text=vcol).attr_name = vcol
         else:
-            l.label("No Vertex Color layers on objects with this material")
+            l.label(text="No Vertex Color layers on objects with this material")
 
 
 class NWSwitchNodeTypeMenu(Menu, NWBase):
