diff -aur blender280/release/scripts/addons_contrib/object_facemap_auto/auto_fmap_widgets.py blender280.new/release/scripts/addons_contrib/object_facemap_auto/auto_fmap_widgets.py
--- blender280/release/scripts/addons_contrib/object_facemap_auto/auto_fmap_widgets.py	2018-12-11 11:44:41.000000000 +0100
+++ blender280.new/release/scripts/addons_contrib/object_facemap_auto/auto_fmap_widgets.py	2018-12-11 11:59:40.996911850 +0100
@@ -323,9 +323,7 @@
         mpr.select = fmap.select
 
         # foo;bar=baz;bonzo=bingo --> {"bar": baz", "bonzo": bingo}
-        mpr.fmap_target_rules = dict(
-            item.partition("=")[::2] for item in fmap_rules
-        )
+        mpr.fmap_target_rules = dict(item.partition("=")[::2] for item in fmap_rules)
 
         # XXX, we might want to have some way to extract a 'center' from a face-map
         # for now use the pose-bones location.
