diff -aur openvdb-6.0.0/openvdb/CMakeLists.txt openvdb-6.0.0.new/openvdb/CMakeLists.txt
--- openvdb-6.0.0/openvdb/CMakeLists.txt	2018-12-18 13:27:52.000000000 +0100
+++ openvdb-6.0.0.new/openvdb/CMakeLists.txt	2018-12-30 10:31:29.041366761 +0100
@@ -275,6 +275,7 @@
   TARGET_LINK_LIBRARIES ( vdb_view
     openvdb_shared
     ${Boost_THREAD_LIBRARY}
+    ${CMAKE_THREAD_LIBS_INIT}
     ${OPENGL_gl_LIBRARY}
     ${OPENGL_glu_LIBRARY}
     ${COCOA_LIBRARY}
