diff -aur linux-3.10/drivers/scsi/scsi.c linux-3.10.new/drivers/scsi/scsi.c
--- linux-3.10/drivers/scsi/scsi.c	2013-08-15 07:39:52.285559467 +0200
+++ linux-3.10.new/drivers/scsi/scsi.c	2013-08-15 07:42:34.239421459 +0200
@@ -1031,6 +1031,9 @@
 {
 	int i, result;
 
+       if (sdev->skip_vpd_pages)
+               goto fail;
+
 	/* Ask for all the pages supported by this device */
 	result = scsi_vpd_inquiry(sdev, buf, 0, buf_len);
 	if (result)
