diff -aur blender/intern/cycles/scene/image_vdb.cpp blender.new/intern/cycles/scene/image_vdb.cpp
--- blender/intern/cycles/scene/image_vdb.cpp	2024-05-19 12:21:31.000000000 +0200
+++ blender.new/intern/cycles/scene/image_vdb.cpp	2024-11-10 14:55:09.161188804 +0100
@@ -12,7 +12,8 @@
 #endif
 #ifdef WITH_NANOVDB
 #  define NANOVDB_USE_OPENVDB
-#  include <nanovdb/util/OpenToNanoVDB.h>
+#     include <nanovdb/util/Util.h>// for NANOVDB_DEPRECATED_HEADER
+#     include <nanovdb/tools/CreateNanoGrid.h>
 #endif
 
 CCL_NAMESPACE_BEGIN
@@ -54,8 +55,8 @@
     if constexpr (!std::is_same_v<GridType, openvdb::MaskGrid>) {
       try {
 #    if NANOVDB_MAJOR_VERSION_NUMBER > 32 || \
-        (NANOVDB_MAJOR_VERSION_NUMBER == 32 && NANOVDB_MINOR_VERSION_NUMBER >= 6)
-        /* OpenVDB 11. */
+        (NANOVDB_MAJOR_VERSION_NUMBER == 32 && NANOVDB_MINOR_VERSION_NUMBER >= 7)
+        /* OpenVDB 11. 12. */
         if constexpr (std::is_same_v<FloatGridType, openvdb::FloatGrid>) {
           openvdb::FloatGrid floatgrid(*openvdb::gridConstPtrCast<GridType>(grid));
           if (precision == 0) {
diff -aur blender/intern/cycles/scene/image_vdb.h blender.new/intern/cycles/scene/image_vdb.h
--- blender/intern/cycles/scene/image_vdb.h	2024-05-19 12:21:31.000000000 +0200
+++ blender.new/intern/cycles/scene/image_vdb.h	2024-11-10 14:49:49.159279456 +0100
@@ -9,7 +9,7 @@
 #  include <openvdb/openvdb.h>
 #endif
 #ifdef WITH_NANOVDB
-#  include <nanovdb/util/GridHandle.h>
+#  include <nanovdb/GridHandle.h>
 #endif
 
 #include "scene/image.h"
