diff -aur blender/intern/cycles/kernel/osl/osl_closures.cpp blender.new/intern/cycles/kernel/osl/osl_closures.cpp
--- blender/intern/cycles/kernel/osl/osl_closures.cpp	2017-04-23 11:21:38.000000000 +0200
+++ blender.new/intern/cycles/kernel/osl/osl_closures.cpp	2017-04-24 11:08:04.640017518 +0200
@@ -242,7 +242,7 @@
 	/* optimization: it's possible to not use a prepare function at all and
 	 * only initialize the actual class when accessing the closure component
 	 * data, but then we need to map the id to the class somehow */
-	ss->register_closure(name, id, params, prepare, NULL, 16);
+	ss->register_closure(name, id, params, prepare, NULL);
 }
 
 void OSLShader::register_closures(OSLShadingSystem *ss_)
