diff -aur Pixie/src/ri/rib.l Pixie.new/src/ri/rib.l
--- Pixie/src/ri/rib.l	2008-11-12 08:00:24.000000000 +0100
+++ Pixie.new/src/ri/rib.l	2012-02-29 21:13:30.873536763 +0100
@@ -17,7 +17,7 @@
 
 // Overwrite the YYinput so that it uses libz
 #undef YY_INPUT
-#define YY_INPUT(buf, retval, maxlen)	if ( (retval = gzread(ribin,buf,maxlen)) < 0) 			\
+#define YY_INPUT(buf, retval, maxlen)	if ( (retval = gzread((gzFile)ribin,buf,maxlen)) < 0) 			\
 											YY_FATAL_ERROR( "input in flex scanner failed" );
 
 #endif
@@ -223,7 +223,7 @@
 															TRibFile	*nextFile	=	ribStack->next;
 															rib_delete_buffer( YY_CURRENT_BUFFER );
 #ifdef HAVE_ZLIB
-															gzclose(ribin);
+															gzclose((gzFile)ribin);
 #else
 															fclose(ribin);
 #endif
@@ -243,7 +243,7 @@
 															rib_delete_buffer( YY_CURRENT_BUFFER );
 															
 #ifdef HAVE_ZLIB
-															gzclose(ribin);
+															gzclose((gzFile)ribin);
 #else
 															fclose(ribin);
 #endif
diff -aur Pixie/src/ri/ribOut.cpp Pixie.new/src/ri/ribOut.cpp
--- Pixie/src/ri/ribOut.cpp	2009-02-23 06:48:03.000000000 +0100
+++ Pixie.new/src/ri/ribOut.cpp	2012-02-29 21:12:31.533535479 +0100
@@ -162,7 +162,7 @@
 
 #ifdef HAVE_ZLIB
 			if (outputCompressed) {
-				gzclose(outFile);
+				gzclose((gzFile)outFile);
 			} else {
 				fclose(outFile);
 			}
diff -aur Pixie/src/ri/ribOut.h Pixie.new/src/ri/ribOut.h
--- Pixie/src/ri/ribOut.h	2009-02-23 06:48:03.000000000 +0100
+++ Pixie.new/src/ri/ribOut.h	2012-02-29 21:11:57.853534631 +0100
@@ -237,7 +237,7 @@
 												const int	l	=	vsprintf(scratch,mes,args);
 
 												#ifdef HAVE_ZLIB
-													if (outputCompressed)	gzwrite(outFile,scratch,l);
+													if (outputCompressed)	gzwrite((gzFile)outFile,scratch,l);
 													else					fwrite(scratch,1,l,outFile);
 												#else
 													fwrite(scratch,1,l,outFile);
@@ -258,7 +258,7 @@
 												const int l	=	vsprintf(scratch,mes,args);
 
 												#ifdef HAVE_ZLIB
-													if (outputCompressed)	gzwrite(outFile,scratch,l);
+													if (outputCompressed)	gzwrite((gzFile)outFile,scratch,l);
 													else					fwrite(scratch,1,l,outFile);
 												#else
 													fwrite(scratch,1,l,outFile);
diff -aur Pixie/src/ri/rib.y Pixie.new/src/ri/rib.y
--- Pixie/src/ri/rib.y	2009-01-27 01:32:50.000000000 +0100
+++ Pixie.new/src/ri/rib.y	2012-02-29 21:14:11.666870977 +0100
@@ -2940,7 +2940,7 @@
 		
 		if (ribin != NULL) {
 #ifdef HAVE_ZLIB
-			gzclose(ribin);
+			gzclose((gzFile)ribin);
 #else
 			fclose(ribin);
 #endif
