diff -aur blender/intern/cycles/util/util_simd.h blender.new/intern/cycles/util/util_simd.h
--- blender/intern/cycles/util/util_simd.h	2015-08-02 10:14:27.000000000 +0200
+++ blender.new/intern/cycles/util/util_simd.h	2015-08-30 09:53:50.937537982 +0200
@@ -67,8 +67,10 @@
 #endif
 
 #if defined(__LZCNT__)
+/* gcc 5.2.0 : redefinition
 #define _lzcnt_u32 __lzcnt32
 #define _lzcnt_u64 __lzcnt64
+*/
 #endif
 
 #if defined(_WIN32) && !defined(__MINGW32__)
