diff --git a/source/blender/collada/DocumentImporter.h b/source/blender/collada/DocumentImporter.h
index 17e61326032..1c204f3e799 100644
--- a/source/blender/collada/DocumentImporter.h
+++ b/source/blender/collada/DocumentImporter.h
@@ -107,6 +107,8 @@ public:
        bool writeAnimation(const COLLADAFW::Animation*);
 
        bool writeAnimationList(const COLLADAFW::AnimationList*);
+       
+       bool writeAnimationClip(const COLLADAFW::AnimationClip*);
 
        bool writeGeometry(const COLLADAFW::Geometry*);
 
diff --git a/source/blender/collada/DocumentImporter.cpp b/source/blender/collada/DocumentImporter.cpp
index 67fd9d648c7..d68b5ceaf80 100644
--- a/source/blender/collada/DocumentImporter.cpp
+++ b/source/blender/collada/DocumentImporter.cpp
@@ -1349,6 +1349,13 @@ bool DocumentImporter::writeAnimationList(const COLLADAFW::AnimationList *animat
        return anim_importer.write_animation_list(animationList);
 }
 
+/** When this method is called, the writer must write the AnimationClip.
+ * /return The writer should return true, if writing succeeded, false otherwise.*/
+bool DocumentImporter::writeAnimationClip(const COLLADAFW::AnimationClip *AnimationClip)
+{
+       return true;
+}
+
 /** When this method is called, the writer must write the skin controller data.
  * \return The writer should return true, if writing succeeded, false otherwise.*/
 bool DocumentImporter::writeSkinControllerData(const COLLADAFW::SkinControllerData *skin)
