diff -aur blender/release/scripts/addons_contrib/np_station/np_point_dimension.py blender.new/release/scripts/addons_contrib/np_station/np_point_dimension.py
--- blender/release/scripts/addons_contrib/np_station/np_point_dimension.py	2017-04-23 11:53:29.000000000 +0200
+++ blender.new/release/scripts/addons_contrib/np_station/np_point_dimension.py	2017-06-25 11:15:52.373892605 +0200
@@ -141,11 +141,8 @@
 
 
 def register():
-
-
+    pass
 
 
 def unregister():
-
-
     pass
