diff -aur Amaya11.4.4/Amaya/amaya/html2thot.c Amaya11.4.4.new/Amaya/amaya/html2thot.c
--- Amaya11.4.4/Amaya/amaya/html2thot.c	2010-07-22 11:54:27.000000000 +0200
+++ Amaya11.4.4.new/Amaya/amaya/html2thot.c	2012-02-19 10:39:24.266186048 +0100
@@ -4349,7 +4349,7 @@
       WorkBuffer = FileBuffer;
     }
   // need to read a new set of characters
-  LastCharInWorkBuffer = gzread (infile, &FileBuffer[StartOfRead],
+  LastCharInWorkBuffer = gzread ((gzFile)infile, &FileBuffer[StartOfRead],
                                  INPUT_FILE_BUFFER_SIZE - StartOfRead);
   // add previous read characters not managed yet
   LastCharInWorkBuffer += StartOfRead;
diff -aur Amaya11.4.4/Amaya/amaya/Xml2thot.c Amaya11.4.4.new/Amaya/amaya/Xml2thot.c
--- Amaya11.4.4/Amaya/amaya/Xml2thot.c	2009-10-12 09:35:20.000000000 +0200
+++ Amaya11.4.4.new/Amaya/amaya/Xml2thot.c	2012-02-19 10:41:10.326188342 +0100
@@ -5847,7 +5847,7 @@
   while (!endOfFile && !XMLNotWellFormed && !XMLInvalidToken)
     {
       /* read the XML file */
-      res = gzread (infile, bufferRead, COPY_BUFFER_SIZE);
+      res = gzread ((gzFile)infile, bufferRead, COPY_BUFFER_SIZE);
       if (res < 0)
         return;
 
