diff -aur mozilla-release/bluegriffon/app/profile/bluegriffon-prefs.js mozilla-release.new/bluegriffon/app/profile/bluegriffon-prefs.js
--- mozilla-release/bluegriffon/app/profile/bluegriffon-prefs.js	2013-05-26 06:31:09.834150363 +0200
+++ mozilla-release.new/bluegriffon/app/profile/bluegriffon-prefs.js	2013-06-09 09:43:01.288415678 +0200
@@ -125,8 +125,8 @@
 pref("bluegriffon.structurebar.role.show", true);
 
 // updates
-pref("bluegriffon.updates.check.enabled", true);
-pref("bluegriffon.updates.frequency", "launch");
+pref("bluegriffon.updates.check.enabled", false);
+pref("bluegriffon.updates.frequency", "manually");
 
 pref("html5.enable", true);
 pref("bluegriffon.defaults.doctype", "kHTML5");
@@ -149,6 +149,11 @@
 pref("bluegriffon.osx.dock-integration", true);
 pref("extensions.getAddons.cache.enabled", false);
 
+// Locales
+pref("general.useragent.locale", "");
+pref("intl.locale.matchOS", true);
+
+
 // +------------------------------------+
 // | Disabled for now, too late for 1.4 |
 // +------------------------------------+
