diff -aur blender/source/blender/render/intern/raytrace/rayobject_rtbuild.cpp blender.new/source/blender/render/intern/raytrace/rayobject_rtbuild.cpp
--- blender/source/blender/render/intern/raytrace/rayobject_rtbuild.cpp	2016-08-14 11:59:28.000000000 +0200
+++ blender.new/source/blender/render/intern/raytrace/rayobject_rtbuild.cpp	2016-08-24 12:37:04.994169493 +0200
@@ -35,16 +35,16 @@
 #include <stdlib.h>
 #include <algorithm>
 
-#include "rayobject_rtbuild.h"
+#ifndef isfinite
+#define isfinite(x) std::isfinite(x)
+#endif
 
 #include "MEM_guardedalloc.h"
 
 #include "BLI_math.h"
 #include "BLI_utildefines.h"
 
-#if __cplusplus >= 201103L
-using std::isfinite;
-#endif
+#include "rayobject_rtbuild.h"
 
 static bool selected_node(RTBuilder::Object *node)
 {
