diff -aur linux-3.1/drivers/ata/libata-core.c linux-3.1.new/drivers/ata/libata-core.c
--- linux-3.1/drivers/ata/libata-core.c	2011-10-24 09:10:05.000000000 +0200
+++ linux-3.1.new/drivers/ata/libata-core.c	2011-11-24 19:56:34.611356618 +0100
@@ -4126,6 +4126,7 @@
 	{ "PIONEER DVD-RW  DVRTD08",	NULL,	ATA_HORKAGE_NOSETXFER },
 	{ "PIONEER DVD-RW  DVR-212D",	NULL,	ATA_HORKAGE_NOSETXFER },
 	{ "PIONEER DVD-RW  DVR-216D",	NULL,	ATA_HORKAGE_NOSETXFER },
+	{ "PIONEER DVD-RW  DVR-216",	"1.07",	ATA_HORKAGE_NOSETXFER },
 
 	/* End Marker */
 	{ }
