diff -aur blender/build_files/cmake/platform/platform_unix.cmake blender.new/build_files/cmake/platform/platform_unix.cmake
--- blender/build_files/cmake/platform/platform_unix.cmake	2025-06-10 16:49:56.000000000 +0200
+++ blender.new/build_files/cmake/platform/platform_unix.cmake	2025-06-15 14:36:48.843189354 +0200
@@ -574,12 +574,13 @@
 endif()
 
 if(WITH_MANIFOLD)
-  if(WITH_LIBS_PRECOMPILED OR WITH_STRICT_BUILD_OPTIONS)
-    find_package(manifold REQUIRED)
+  find_package_wrapper(manifold)
+  if(manifold_FOUND)
+    get_target_property(MANIFOLD_LIBRARIES manifold::manifold LOCATION)
+    get_target_property(MANIFOLD_INCLUDE_DIR manifold::manifold INTERFACE_INCLUDE_DIRECTORIES)
+    message(STATUS "Found manifold: ${MANIFOLD_LIBRARIES}")
   else()
-    # This isn't a common system library, so disable if it's not found.
-    find_package(manifold)
-    set_and_warn_library_found("MANIFOLD" MANIFOLD_FOUND WITH_MANIFOLD)
+    message(STATUS "manifold not found")
   endif()
 endif()
 
