diff -aur blender/build_files/cmake/platform/platform_unix.cmake blender.new/build_files/cmake/platform/platform_unix.cmake
--- blender/build_files/cmake/platform/platform_unix.cmake	2025-05-03 12:38:54.000000000 +0200
+++ blender.new/build_files/cmake/platform/platform_unix.cmake	2025-05-03 15:33:27.507410393 +0200
@@ -596,7 +596,7 @@
     find_package(manifold REQUIRED)
   else()
     # This isn't a common system library, so disable if it's not found.
-    find_package(manifold)
+    find_package_wrapper(manifold)
     set_and_warn_library_found("MANIFOLD" MANIFOLD_FOUND WITH_MANIFOLD)
   endif()
 endif()
