diff -aur oiio-Release-1.8.16/src/cmake/modules/FindOpenCV.cmake oiio-Release-1.8.16.new/src/cmake/modules/FindOpenCV.cmake
--- oiio-Release-1.8.16/src/cmake/modules/FindOpenCV.cmake	2018-11-01 06:35:01.000000000 +0100
+++ oiio-Release-1.8.16.new/src/cmake/modules/FindOpenCV.cmake	2018-11-27 08:10:59.527035968 +0100
@@ -16,7 +16,7 @@
    SET(LibRaw_r_DEFINITIONS ${PC_LIBRAW_R_CFLAGS_OTHER})   
 ENDIF()
 
-find_path (OpenCV_INCLUDE_DIR opencv/cv.h
+find_path (OpenCV_INCLUDE_DIR opencv2/cvconfig.h
            "${PROJECT_SOURCE_DIR}/include"
            "${OpenCV_DIR}/include"
            "$ENV{OpenCV_DIR}/include"
