diff -aur oiio-Release-1.8.16/src/libOpenImageIO/imagebufalgo_opencv.cpp oiio-Release-1.8.16.new/src/libOpenImageIO/imagebufalgo_opencv.cpp
--- oiio-Release-1.8.16/src/libOpenImageIO/imagebufalgo_opencv.cpp	2018-11-01 06:35:01.000000000 +0100
+++ oiio-Release-1.8.16.new/src/libOpenImageIO/imagebufalgo_opencv.cpp	2018-11-27 08:18:01.492576613 +0100
@@ -201,9 +201,9 @@
 
     // OpenCV uses BGR ordering
     if (spec.nchannels == 3) {
-        cvCvtColor(ipl, ipl, CV_RGB2BGR);
+        cvCvtColor(ipl, ipl, cv::COLOR_RGB2BGR);
     } else if (spec.nchannels == 4) {
-        cvCvtColor(ipl, ipl, CV_RGBA2BGRA);
+        cvCvtColor(ipl, ipl, cv::COLOR_RGBA2BGRA);
     }
 
     return ipl;
