diff -aur OpenImageIO-2.5.18.0/src/psd.imageio/psdinput.cpp OpenImageIO-2.5.18.0.new/src/psd.imageio/psdinput.cpp
--- OpenImageIO-2.5.18.0/src/psd.imageio/psdinput.cpp	2025-01-01 21:04:08.000000000 +0100
+++ OpenImageIO-2.5.18.0.new/src/psd.imageio/psdinput.cpp	2025-02-07 11:15:49.178123832 +0100
@@ -2234,6 +2234,7 @@
     int16_t header;
     int length;
 
+    char* dst_start = dst;
     while (src_remaining > 0 && dst_remaining > 0) {
         header = *reinterpret_cast<const signed char*>(src);
         src++;
@@ -2276,8 +2277,8 @@
 
     if (!bigendian()) {
         switch (m_header.depth) {
-        case 16: swap_endian((uint16_t*)dst, m_spec.width); break;
-        case 32: swap_endian((uint32_t*)dst, m_spec.width); break;
+        case 16: swap_endian((uint16_t*)dst_start, m_spec.width); break;
+        case 32: swap_endian((uint32_t*)dst_start, m_spec.width); break;
         }
     }
 
