diff -aur openvdb-8.0.1/openvdb/openvdb/cmd/CMakeLists.txt openvdb-8.0.1.new/openvdb/openvdb/cmd/CMakeLists.txt
--- openvdb-8.0.1/openvdb/openvdb/cmd/CMakeLists.txt	2021-02-05 20:21:38.000000000 +0100
+++ openvdb-8.0.1.new/openvdb/openvdb/cmd/CMakeLists.txt	2021-04-29 10:50:18.260123118 +0200
@@ -116,8 +116,8 @@
 #### vdb_render
 
 if(OPENVDB_BUILD_VDB_RENDER)
-  find_package(IlmBase ${MINIMUM_ILMBASE_VERSION} REQUIRED COMPONENTS Half Iex IlmThread Imath)
-  find_package(OpenEXR ${MINIMUM_OPENEXR_VERSION} REQUIRED COMPONENTS IlmImf)
+  find_package(Imath CONFIG REQUIRED)
+  find_package(OpenEXR CONFIG REQUIRED)
 
   set(VDB_RENDER_SOURCE_FILES openvdb_render.cc)
   add_executable(vdb_render ${VDB_RENDER_SOURCE_FILES})
@@ -129,10 +129,8 @@
   # headers
 
   target_link_libraries(vdb_render
-    OpenEXR::IlmImf
-    IlmBase::IlmThread
-    IlmBase::Iex
-    IlmBase::Imath
+    OpenEXR::OpenEXR
+    Imath::Imath
     ${OPENVDB_BINARIES_DEPENDENT_LIBS}
   )
 
