diff -aur blender/scripts/addons_core/io_scene_gltf2/io/com/draco.py blender.new/scripts/addons_core/io_scene_gltf2/io/com/draco.py
--- blender/scripts/addons_core/io_scene_gltf2/io/com/draco.py	2024-11-03 10:38:55.000000000 +0100
+++ blender.new/scripts/addons_core/io_scene_gltf2/io/com/draco.py	2024-11-03 10:45:54.864192283 +0100
@@ -3,6 +3,7 @@
 # SPDX-License-Identifier: Apache-2.0
 
 import os
+import site
 import sys
 from pathlib import Path
 import bpy
@@ -35,7 +36,7 @@
     python_lib = Path('{v[0]}.{v[1]}/python/lib'.format(v=bpy.app.version))
     python_version = 'python{v[0]}.{v[1]}'.format(v=sys.version_info)
 
-    path = os.environ.get('BLENDER_EXTERN_DRACO_LIBRARY_PATH')
+    path = f"{site.getsitepackages()[0]}/libextern_draco.so"
     if path is not None:
         return Path(path)
 
