--- ./src/VTK/IO/vtkPNGReader.cxx.orig	2010-02-04 01:49:09.000000000 +0100
+++ ./src/VTK/IO/vtkPNGReader.cxx	2010-02-04 01:52:56.000000000 +0100
@@ -116,7 +116,7 @@
   // minimum of a byte per pixel
   if (color_type == PNG_COLOR_TYPE_GRAY && bit_depth < 8) 
     {
-    png_set_gray_1_2_4_to_8(png_ptr);
+    png_set_expand_gray_1_2_4_to_8(png_ptr);
     }
 
   // add alpha if any alpha found
@@ -225,7 +225,7 @@
   // minimum of a byte per pixel
   if (color_type == PNG_COLOR_TYPE_GRAY && bit_depth < 8) 
     {
-    png_set_gray_1_2_4_to_8(png_ptr);
+    png_set_expand_gray_1_2_4_to_8(png_ptr);
     }
 
   // add alpha if any alpha found
