diff -aur mozilla-1.9.2/gfx/thebes/public/gfxPDFSurface.h mozilla-1.9.2.new/gfx/thebes/public/gfxPDFSurface.h
--- mozilla-1.9.2/gfx/thebes/public/gfxPDFSurface.h	2010-01-21 04:30:13.000000000 +0100
+++ mozilla-1.9.2.new/gfx/thebes/public/gfxPDFSurface.h	2010-01-23 11:45:46.174898111 +0100
@@ -63,7 +63,11 @@
     // this is in points!
     const gfxSize& GetSize() const { return mSize; }
 
-    virtual PRInt32 GetDefaultContextFlags() const { return gfxContext::FLAG_DISABLE_SNAPPING; }
+    virtual PRInt32 GetDefaultContextFlags() const
+    {
+        return gfxContext::FLAG_SIMPLIFY_OPERATORS |
+               gfxContext::FLAG_DISABLE_SNAPPING;
+    }
 
 private:
     nsCOMPtr<nsIOutputStream> mStream;
diff -aur mozilla-1.9.2/gfx/thebes/public/gfxPSSurface.h mozilla-1.9.2.new/gfx/thebes/public/gfxPSSurface.h
--- mozilla-1.9.2/gfx/thebes/public/gfxPSSurface.h	2010-01-21 04:30:13.000000000 +0100
+++ mozilla-1.9.2.new/gfx/thebes/public/gfxPSSurface.h	2010-01-23 11:46:45.258638319 +0100
@@ -63,7 +63,11 @@
     // this is in points!
     const gfxSize& GetSize() const { return mSize; }
 
-    virtual PRInt32 GetDefaultContextFlags() const { return gfxContext::FLAG_DISABLE_SNAPPING; }
+    virtual PRInt32 GetDefaultContextFlags() const
+    {
+        return gfxContext::FLAG_SIMPLIFY_OPERATORS |
+               gfxContext::FLAG_DISABLE_SNAPPING;
+    }
 
 private:
     nsCOMPtr<nsIOutputStream> mStream;
